x-pack/plugins/enterprise_search/public/applications/shared/hidden_text/hidden_text.tsx

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, ReactElement } from 'react';
+
+interface ChildrenProps {
+  toggle: () => void;
+  isHidden: boolean;
+  hiddenText: string;
+}
+
+interface Props {
+  text: string;
+  children(props: ChildrenProps): ReactElement;
+}
+
+export const HiddenText: React.FC<Props> = ({ text, children }) => {
+  const [isHidden, toggleIsHidden] = useState(true);
+  const hiddenText = isHidden ? text.replace(/./g, '*') : text;
```

Comment:
Per my previous collapsed comment - I'd still love to bake a minor accessibility improvement into this component if possible!

```suggestion

  const hiddenLabel = i18n.translate('xpack.enterpriseSearch.hiddenText', {
    defaultMessage: 'Hidden text',
  });
  const hiddenText = isHidden ? (
    <span aria-label={hiddenLabel}>{text.replace(/./g, '•')}</span>
  ) : (
    text
  );
```

You'll have to update the type defs of `hiddenText` and `Key.text` to `React.ReactNode` (accepts either a string or node) as a heads up. I have a diff here if that helps: 4de85a6 (I didn't update tests in that one though, apologies!)

Reply:
This is cool, I hadn't considered this.

Reply:
8377454e82bd1715c8f53422014ac0b9a6523149